### DISCUSSION TITLE
Setting Prebuilt Blocks (Template) into the PageBuilder

### DISCUSSION THREAD

**Discussion by Seyikins27** [Z3d0X/filament-fabricator#84] [Q&A] [answered]
I have a Template Module that uses the FilamentFabricator Package to pre-arrange different page blocks into a template that can be reused several times in the PageResource. I do this by Using a select field which has a list of the templates to set the 'blocks' PageBuilder field. This works perfectly until i load a template that uses a file upload component or a Package called FilamentCurator, then it throws this error "foreach() argument must be of type array|object, string given" or "count(): Argument #1 ($value) must be of type Countable|array, int given". Please help out with this. Below is a code snippet of the PageResource and how i use the template

```
 Group::make()
                ->schema([
                    Select::make('templates')
                    ->label('Load From Existing Templates')
                    ->options(function (){
                       return Template::get_template()->pluck('name','id');
                    })
                    ->reactive()
                    ->preload()
                    ->searchable()
                    ->afterStateUpdated(function (Closure $set, $state, Component $component) {
                        $set('blocks', function() use($state, $component){
                            if(blank($state))
                            {
                                return [];
                            }
                            else
                            {
                                $template=Template::find($state);
                                return $template->blocks;
                            }

                        });
                    }),
                    Group::make()->schema(FilamentFabricator::getSchemaSlot('blocks.before')),
                    PageBuilder::make('blocks')
                        ->label(__('filament-fabricator::page-resource.labels.blocks'))->collapsible(),
                    Group::make()->schema(FilamentFabricator::getSchemaSlot('blocks.after')),
                ])
                ->columnSpan(2),
```
![Screenshot_page](https://github.com/Z3d0X/filament-fabricator/assets/30042271/f5bed597-41f5-495a-8c2d-92938e91f86f)

![Screenshot_error](https://github.com/Z3d0X/filament-fabricator/assets/30042271/0dc128fe-43f0-4760-9684-2a81b00d491b)

![Screenshot_error_2](https://github.com/Z3d0X/filament-fabricator/assets/30042271/797b598c-5c91-4b7b-8c99-25b4b0617c30)

**Comment by Z3d0X** [comment, marked as answer]
I don't think you can use `$set` directly on a FileUpload input. Instead try something like this:
```php
->afterStateUpdated(function ($state, $livewire) {
    if(blank($state)) {
        $blocks = null;
    } else {
        $template = Template::find($state);
        $blocks = $template->blocks;
    }

    $newState = [
        ...$livewire->form->getRawState(),
        'blocks' => $blocks,
    ];

    $livewire->form->fill($newState);
})
```

**Reply by Seyikins27** [reply]
Thank you so much @Z3d0X , this really helped to answer my question
